liquidity/liquidity.go

```diff
@@ -386,7 +450,33 @@ func (m *Manager) SuggestSwaps(ctx context.Context) (
 		suggestions = append(suggestions, outRequest)
 	}
 
-	return suggestions, nil
+	// If we have no suggestions after we have applied all of our limits,
+	// just return.
+	if len(suggestions) == 0 {
+		return nil, nil
+	}
+
+	// Sort suggestions by amount in descending order.
+	sort.SliceStable(suggestions, func(i, j int) bool {
+		return suggestions[i].Amount > suggestions[j].Amount
+	})
+
+	// Run through our suggested swaps in descending order of amount and
+	// return the first swap that will fall beneath our fee budget if it
+	// succeeds.
+	for _, swap := range suggestions {
+		successFees := swap.MaxMinerFee + swap.MaxSwapFee +
+			swap.MaxPrepayRoutingFee + swap.MaxSwapRoutingFee
+
+		if spent+successFees <= m.params.AutoFeeBudget {
```

Comment:
I'm not sure I understand why only one swap is suggested here.
Wouldn't we want to suggest _all_ of the swaps that can fit in our remaining budget? Or is this a safety measure to not overshoot the budget by only suggesting one at a time?

Reply:
I was thinking that it's safer to have a single swap at a time, but have updated this to a `MaxInFlight` param so that it can be toggled as people like. Since we only do one swap per channel at a time, and use absolute worst case fees, I think it's safe to give people the option of multiple autoloops at once? Esp since sweeps could take a very long time for loop out.

Reply:
Yeah, makes sense :+1: